geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/connection/AuthIntegrationTest.java

```diff
@@ -227,4 +244,85 @@ public void givenSecurityWithWritePermission_getCommandFails() throws Exception
     assertThatThrownBy(() -> jedis.get("foo"))
         .hasMessageContaining(RedisConstants.ERROR_NOT_AUTHORIZED);
   }
+
+  @Test
+  public void givenSecurity_authCommandPasswordIsNotLoggedAtDebugLevel() throws IOException {
+    File logFile = temporaryFolder.newFile();
+
+    createRadishServerWithLogFileAndSecurityManager(logFile, "fine", NoOpSecurityManager.class);
+
+    jedis.auth(LOGGED_PASSWORD);
+
+    checkLogFileForPassword(logFile, LOGGED_PASSWORD);
+  }
+
+  @Test
+  public void givenSecurity_authCommandPasswordIsNotLoggedOnException() throws IOException {
+    File logFile = temporaryFolder.newFile();
+
+    createRadishServerWithLogFileAndSecurityManager(logFile, "info",
+        ThrowsOnAuthorizeSecurityManager.class);
+
+    // The first AUTH command will authenticate the user, and the second will cause the authorize()
+    // method on the ThrowsOnAuthorizeSecurityManager to be invoked, throwing an exception
+    jedis.auth(LOGGED_PASSWORD);
+    assertThatThrownBy(() -> jedis.auth(LOGGED_PASSWORD))
+        .hasMessageContaining(SERVER_ERROR_MESSAGE);
+
+    checkLogFileForPassword(logFile, LOGGED_PASSWORD);
+  }
+
+  private void createRadishServerWithLogFileAndSecurityManager(File logFile, String logLevel,
+      Class<?> securityManager) {
+    System.setProperty("io.netty.eventLoopThreads", "1");
+    port = AvailablePortHelper.getRandomAvailableTCPPort();
```

Comment:
I think it would be safer to move this down one line. I think there's a possibility that the cache creation would end up using the port just deemed free here.

Reply:
Moved